nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-processors/src/main/java/org/apache/nifi/processors/kafka/KafkaPublisher.java

```diff
@@ -114,10 +121,13 @@ KafkaPublisherResult publish(PublishingContext publishingContext) {
 
         int prevLastAckedMessageIndex = publishingContext.getLastAckedMessageIndex();
         List<Future<RecordMetadata>> resultFutures = new ArrayList<>();
+        System.out.println(1 - prevLastAckedMessageIndex);
 
         byte[] messageBytes;
         int tokenCounter = 0;
-        for (; (messageBytes = streamTokenizer.nextToken()) != null; tokenCounter++) {
+        boolean continueSending = true;
+        KafkaPublisherResult result = null;
+        for (; (messageBytes = streamTokenizer.nextToken()) != null && continueSending; tokenCounter++) {
```

Comment:
Should we change the conditional here to `continueSending && (messageBytes = streamTokenizer.nextToken()) != null` so that we don't bother calling streamTokenizer.nextToken() if `continueSending` is false?